docs/content/basic/custom-steps.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Listening and responding to custom steps
+lang: en
+slug: /concepts/custom-steps
+---
+
+Your app can use the `function()` method to listen to incoming [custom step requests](https://api.slack.com/automation/functions/custom-bolt). Custom steps are used in Workflow Builder to build workflows. The method requires a step `callback_id` of type string. This `callback_id` must also be defined in your [Function](https://api.slack.com/concepts/manifests#functions) definition. Custom steps must be finalized using the `complete()` or `fail()` listener arguments to notify Slack that your app has processed the request.
+
+* `complete()` requires one argument: an `outputs` object. It completes your custom step **successfully** and provides an object containing the outputs of your custom step as per its definition.
+* `fail()` requires **one** argument: `error` of type string. It completes your custom step **unsuccessfully** and provides a message containing information regarding why your custom step failed.
+
+You can reference your custom step's inputs using the `inputs` listener argument.
+
+```js
+// This sample custom step formats an input and outputs it
+app.function('sample_custom_step', async ({ ack, inputs, complete, fail, logger }) => {
+  try {
+    await ack();
```

Comment:
```suggestion
app.function('sample_custom_step', async ({ inputs, complete, fail, logger }) => {
  try {
```

Removing the `ack` here since this isn't provided for function executions

Reply:
I prefer to keep it in the docs, there are a number of issues opened related to calling ack before executing custom logic

Keeping this in hopes that it will help developers avoid a bug 🙏

We can remove it in a follow up PR as well 🤔

Reply:
I believe the `function_executed` events are actually different from other events and can't use `ack` at all!

Within JavaScript code this is `undefined` within the listener so this call will error 😬  It's not so clear from this picture but the hover hint is hinting at this in a TypeScript project:

![undefined](https://github.com/user-attachments/assets/a05d9589-c90c-404d-8878-9a7dbd5d536a)